### DISCUSSION TITLE
Ошибка "ярлыка"

### DISCUSSION THREAD

**Discussion by mihail-cornev** [totumonline/totum-ru-issues-and-discussions#459] [Как это сделать?] [answered]
В видео описал ошибку. Но напишу текстом:
В таблице циклов создал поле типа Ярлык, которые ссылается на поле типа Селект в расчётной таблице внутри цикла. Выдаёт вот такую ошибку:

> Поле h_material не найдено. (поле finish таблицы products)

А вот код селекта в поле h_finish:
`= : selectRowListForSelect(table: 'products_finishes'; field: 'name_ru_abbr'; where: 'id' = $listadd; order: 'name_ru' asc)
	listadd: listAdd(list: $list; item: $#nfv)
		list: selectList(table: 'products_materials_finishes'; field: 'finish'; where: 'material' = #h_material)`

[ошибка-ярлык.webm](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/113771444/d5be382b-ad34-4c8d-ae02-680647fc214a)

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Селект на который ссылается поле использует поле ... для расчета и его в таблице циклов нет. Ярлыки в таком случае являются плохой архитектурой. Ипользуйте простой Селект вместо Ярлык.

**Reply by mihail-cornev** [reply]
Спасибо, понял. Хотелось как проще ) видимо нужно ещё раз учебный курс проштудировать.
